liquidhaskell-boot/src/Language/Haskell/Liquid/Bare/Axiom.hs

```diff
@@ -346,6 +354,26 @@ makeAssumeType allowTC tce lmap dm sym mbT v def
     ref        = F.Reft (F.vv_, F.PAtom F.Eq (F.EVar F.vv_) le)
     mkErr s    = ErrHMeas (sourcePosSrcSpan $ loc sym) (pprint $ val sym) (PJ.text s)
     bbs        = filter isBoolBind xs
+
+    -- Before this modification reflected functions that were polymorphic were
+    -- getting compiled to monomorphic fixpint code.  As an example: 
+    -- id :: a -> a ... id x = x 
+    -- Was generating: 
+    -- define id (x : a#foobar) : a#foobar = { (x : a#foobar) }
+    -- Using FObj instead of a real type variable (FVar) This code solves the
+    -- issue by creating a sobritution that replaces those "fake" type variables
+    -- with actual ones.
+    -- define id (x : @-1) : a@-1 = { (x : a@-1) }
```

Comment:
Maybe your code renders the function [eqToPolySort](https://github.com/ucsd-progsys/liquidhaskell/blob/0c88b710a0ac7fc20363527ff88224b221ff923d/liquidhaskell-boot/src/Language/Haskell/Liquid/Constraint/Init.hs#L225-L234) unnecessary, which is only used to substitute sorts in equations as you are doing here.

Reply:
No idea if somewhere we are still generating "fake" type variables

Reply:
Still i think that function can lead to bugs as its using positive type variables indices

Reply:
> No idea if somewhere we are still generating "fake" type variables

It would be a matter of removing the function and checking if the tests pass. I only added it because I had some failing test.

Reply:
Yeah but I believe that on the end we should aim to get rid of that function anyways we should generate correct stuff in the first place

Reply:
I contributed the removal of the substitutions in `eqToPolySort` as they were no longer needed :shrug: